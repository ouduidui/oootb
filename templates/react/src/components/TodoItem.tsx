import React from 'react'
import type { TodoType } from '../stores/todolist'

interface PropsType {
  todo: TodoType
  doneHandle: (id: number) => void
}

export default observer(({ todo, doneHandle }: PropsType) => {
  const navigate = useNavigate()
  const { id, done, content } = todo

  const toggleDone = (e: React.MouseEvent) => {
    e.stopPropagation()
    doneHandle(id)
  }

  return (
    <div
      onClick={() => navigate(`/detail/${id}`)}
      className={`mb-3 px-4 py-2 w-300px border border-gray-200 border-dark:gray-700 bg-transparent flex items-center hover:shadow-sm ${done ? 'opacity-50' : ''}`}>
      <div
        onClick={e => toggleDone(e)}
        className={`cursor-pointer mr-3 ${done ? 'i-carbon-checkbox-checked' : 'i-carbon-checkbox'}`} />
      <div>{content}</div>
    </div>
  )
})
