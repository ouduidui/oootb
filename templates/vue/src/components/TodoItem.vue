<script setup lang="ts">
import type { TodoType } from '~/stores'

const props = defineProps<{
  todo: TodoType
  done?: boolean
}>()

const emits = defineEmits<{
  doneHandle: (id: number, done: boolean) => void
}>()

const router = useRouter()

const toDetail = () => router.push(`/detail/${props.todo.id}`)

</script>

<template>
  <div
    mb-3
    p="x-4 y-2"
    w-300px
    border="~ gray-700 dark:gray-200"
    bg-transparent
    flex
    items-center
    hover:shadow-sm
    :class="props.done ? 'opacity-50' : ''"
    @click="toDetail"
  >
    <div
      cursor-pointer
      mr-3
      :class="props.done ? 'i-carbon-checkbox-checked' : 'i-carbon-checkbox' "
      @click.stop="emits('doneHandle', props.todo.id, !props.done)"
    />
    <div>
      {{ props.todo.content }}
    </div>
  </div>
</template>

<style scope></style>
