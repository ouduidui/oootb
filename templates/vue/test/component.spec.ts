import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import TodoItem from '../src/components/TodoItem.vue'

describe('TodoItem.vue', () => {
  it('should render', () => {
    const wrapper = mount(TodoItem, {
      props: {
        todo: {
          id: 1,
          content: 'Test Todo',
        },
      },
    })
    expect(wrapper.text()).toContain('Test Todo')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
