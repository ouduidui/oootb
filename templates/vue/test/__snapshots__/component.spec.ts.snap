// Vitest Snapshot v1

exports[`HelloWorld.vue > should render 1`] = `
"<div mb-3=\\"\\" p=\\"x-4 y-2\\" w-300px=\\"\\" border=\\"~ gray-700 dark:gray-200\\" bg-transparent=\\"\\" flex=\\"\\" items-center=\\"\\" hover:shadow-sm=\\"\\" class=\\"\\">
  <div cursor-pointer=\\"\\" mr-3=\\"\\" class=\\"i-carbon-checkbox\\"></div>
  <div>Test Todo</div>
</div>"
`;

exports[`TodoItem.vue > should render 1`] = `
"<div mb-3=\\"\\" p=\\"x-4 y-2\\" w-300px=\\"\\" border=\\"~ gray-700 dark:gray-200\\" bg-transparent=\\"\\" flex=\\"\\" items-center=\\"\\" hover:shadow-sm=\\"\\" class=\\"\\">
  <div cursor-pointer=\\"\\" mr-3=\\"\\" class=\\"i-carbon-checkbox\\"></div>
  <div>Test Todo</div>
</div>"
`;
